npm/webpack-dev-server-fresh/src/helpers/createReactAppHandler.ts

```diff
@@ -1,6 +1,428 @@
+import type { CreateFinalWebpackConfig } from '../createWebpackDevServer'
+import debugLib from 'debug'
+import type { Configuration, ResolvePluginInstance, RuleSetRule } from 'webpack'
+import path from 'path'
+import * as globby from 'globby'
+
+type PresetHandlerOptions = Omit<CreateFinalWebpackConfig, 'frameworkConfig'>
+
+const debug = debugLib('cypress:webpack-dev-server-fresh:create-react-app')
+
 /**
  * Sourcing the config for create-react-app
  */
-export function createReactAppHandler () {
-  //
+export function createReactAppHandler (presetHandler: PresetHandlerOptions) {
+  const { devServerConfig, sourceWebpackModulesResult } = presetHandler
+
+  // this is required because
+  // 1) we use our own HMR and we don't need react-refresh transpiling overhead
+  // 2) it doesn't work with process.env=test @see https://github.com/cypress-io/cypress-realworld-app/pull/832
+  process.env.FAST_REFRESH = 'false'
+  const webpackConfig = loadWebpackConfig(presetHandler)
+
+  addCypressToWebpackEslintRulesInPlace(webpackConfig)
+
+  getTranspileFolders(devServerConfig.cypressConfig).forEach((folder) => {
+    allowModuleSourceInPlace(folder, webpackConfig)
+    addFolderToBabelLoaderTranspileInPlace(folder, webpackConfig)
+  })
+
+  if (sourceWebpackModulesResult.webpack.majorVersion === 5) {
+    debug('Modifying configuration for react-scripts@5')
+
+    reactScriptsFiveModifications(webpackConfig)
+  }
+
+  return webpackConfig
+}
+
+function loadWebpackConfig ({ devServerConfig }: PresetHandlerOptions): Configuration {
+  let webpackConfigPath: string
+
+  const envName = 'test'
+
+  process.env.NODE_ENV = envName
+  process.env.BABEL_ENV = envName
+
+  try {
+    // Search for react-scripts webpack config
+    webpackConfigPath = require.resolve('react-scripts/config/webpack.config.js', {
+      paths: [devServerConfig.cypressConfig.projectRoot],
+    })
+
+    debug('Found react-scripts webpack config at %s', webpackConfigPath)
+  } catch (err) {
+    // Might be an ejected cra app, search for common webpack configs
+    const possibleConfigs = globby.sync('**/webpack.config*.js', {
```

Comment:
What's the reason for `**/webpack.config*.js`? (as opposed to `**/webpack.config.js`, or even `config/webpack.config.js`). We should know the location of the ejected CRA webpack config, not sure we should support anything outside the default ejected location?